src/ForkTsCheckerWebpackPluginOptions.json

```diff
@@ -41,8 +41,15 @@
       "required": ["type"]
     },
     "FormatterType":  {
-      "type": "string",
-      "enum": ["basic", "codeframe"]
+      "oneOf": [
+        {
+          "type": "string",
+          "enum": ["basic", "codeframe"]
+        },
+        {
+          "$ref": "#/definitions/FormatterType"
```

Comment:
you can use `"instanceof": "Function"` instead - it's not a standard json schema, but it's supported by schema-utils package that is used for validation plugin options: https://github.com/webpack/schema-utils#examples

Reply:
Update. I did not get the idea of the _it's not a standard json schema_.

The _it's not a standard json schema_ points on `"$ref": "#/definitions/FormatterType"` and `"instanceof": "Function"` or just `"$ref": "#/definitions/FormatterType"`? Im interested on the usage of the `josn schema`. Thanks :D.

Reply:
it's related to the `"instanceof"` - there is no instanceof in json schema spec as far as I know. And it makes sense, because you can't have function or custom class in a JSON, so it's more like a JavaScript related extension of the json schema spec :)